config/unicorn.conf.rb

```diff
@@ -270,9 +291,14 @@ def master_sleep(sec)
 after_fork do |server, worker|
   DiscourseEvent.trigger(:web_fork_started)
   Discourse.after_fork
+  SignalTrapLogger.instance.after_fork
 
-  Signal.trap("USR2") { puts <<~MSG }
-      [#{Time.now.utc.strftime("%Y-%m-%dT%H:%M:%S.%6N")} ##{Process.pid}] Received USR2 signal, dumping backtrace for all threads
-      #{Thread.list.map { |t| "#{t.backtrace&.join("\n")}" }.join("\n\n")}
+  Signal.trap("USR2") do
+    message = <<~MSG
+    Unicorn worker received USR2 signal indicating it is about to timeout, dumping backtrace for main thread
+    #{Thread.current.backtrace&.join("\n")}
     MSG
+
+    SignalTrapLogger.instance.log(Rails.logger, message, level: :warn)
```

Comment:
Note that we're logging to `Rails.logger` here instead of `server.logger` so that the backtraces are easily accessible in `/logs` and the Rails log file.